trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/CegarNwaWorkerThread.java

```diff
@@ -0,0 +1,578 @@
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+			mMainThread.reportFailedContinuesWorkerThread();
+		};
+		Thread.currentThread().setUncaughtExceptionHandler(exhandler);
+
+	}
+
+	/*
+	 * Gets a counterexamples from the blocking queue, sets up a strategy (checks how often the pathprogram has been
+	 * seen). Checks feasibility, interpolates, creates an Error or Interpolant automaton. Calculates the difference to
+	 * generalize the interpolant automaton and then puts a @WorkerThreadResult into the blocking queue for results.
+	 *
+	 * Terminates if the Thread is interrupted (not used atm) or the Executioner service triggers a shutdown.
+	 */
+	@Override
+	public void run() {
+		while (!Thread.currentThread().isInterrupted()) {
+			try {
+				mLogger.info("WorkerThread: " + Thread.currentThread() + " is Waiting for a Task");
+				mIteration += 1;
+				mCounterexample = mNwaCexTransferrer.transferRun((NestedRun) mWorkerTaskQueue.take(), Mode.MAIN2WORKER);
+				final List<L> trace = mCounterexample.getWord().asList();
+				mCurrentErrorLoc = mCounterexample.getSymbol(mCounterexample.getLength() - 2).getTarget();
+				final int traceHash = trace.hashCode();
+				mLogger.info("Starting Thread: " + Thread.currentThread().getId() + "# for Trace Check: " + traceHash);
+				Thread.currentThread().setName("Worker for " + traceHash);
+				try {
+					final var locations = getControlConfigurationsFromCounterexample(mCounterexample);
+					final Counterexample<L> counterexample = new Counterexample<>(mCounterexample.getWord(), locations);
+					final ITARefinementStrategy<L> strategy = setUpStrategy(counterexample);
+					final Pair<LBool, IProgramExecution<L, Term>> isCexResult = isCounterexampleFeasible(strategy);
+
+					final AbstractCegarLoop.AutomatonType automatonType = processFeasibilityCheckResult(strategy,
+							isCexResult.getFirst(), isCexResult.getSecond(), mCurrentErrorLoc);
+					constructRefinementAutomaton(automatonType);
+					mThreadResult = refineAbstractionInternally();
+				} catch (AutomataLibraryException | ToolchainCanceledException | SMTLIBException e) {
+					throw new AssertionError("WorkerThread Failed: " + e);
+				}
+				mLogger.info("Done with Thread: " + Thread.currentThread().getId() + "#");
+				mBlockingQueueForResults.put(mThreadResult);
+			} catch (final InterruptedException e) {
+				Thread.currentThread().interrupt();
+			}
+		}
+	}
+
+	protected List<?> getControlConfigurationsFromCounterexample(final IRun<L, ?> run) {
+		return getIcfgLocationsFromRun(run);
+	}
+
+	private List<IcfgLocation> getIcfgLocationsFromRun(final IRun<L, ?> run) {
+		return run.getStateSequence().stream().map(p -> ((ISLPredicate) p).getProgramPoint())
+				.collect(Collectors.toList());
+	}
+
+	/*
+	 * The worker creates for each counterexample a new strategy module. We can switch between different
+	 * RefinementStrateies here. For example we can use @RefinementStrategy.ACCELERATED_TRACE_CHECK if we have seen the
```

Comment:
nitpick: the javadoc syntax is `{@link RefinementStrategy.ACCELERATED_TRACE_CHECK}`, not `@RefinementStrategy.ACCELERATED_TRACE_CHECK`